src/braket/experimental/autoqasm/api.py

```diff
@@ -351,6 +350,20 @@ def _convert_subroutine(
                 _wrap_for_oqpy_subroutine(_clone_function(f), options)
             )
 
+            # todo: need to validate in other codepaths too?
+            # Validate and declare used qubits
+            quantum_indices = {
+                i
+                # todo: is there a better way to access argument types?
+                for i, arg in enumerate(oqpy_sub.subroutine_declaration[1].arguments)
+                if isinstance(arg, qasm_ast.QuantumArgument)
```

Comment:
Is there a reason you need to get these from the `oqpy` subroutine declaration? What if you just look at the signature of the (Python) function returned from `_wrap_for_oqpy_subroutine`, or even just the signature of the (Python) function `f`?

Reply:
Yeah, I agree, just trying to think through how this will work when we abstract `oqpy` out of this code. I imagine we will end up using the Python signatures here.